backend/data/src/main/kotlin/io/tolgee/service/notification/NotificationSettingService.kt

```diff
@@ -0,0 +1,62 @@
+package io.tolgee.service.notification
+
+import io.tolgee.model.UserAccount
+import io.tolgee.model.notifications.Notification
+import io.tolgee.model.notifications.NotificationChannel
+import io.tolgee.model.notifications.NotificationSetting
+import io.tolgee.model.notifications.NotificationTypeGroup
+import io.tolgee.repository.notification.NotificationSettingRepository
+import org.springframework.stereotype.Service
+
+@Service
+class NotificationSettingService(
+  private val notificationSettingRepository: NotificationSettingRepository,
+) {
+  fun getSettings(user: UserAccount): List<NotificationSetting> {
+    val dbData = notificationSettingRepository.findByUserId(user.id)
+
+    return NotificationTypeGroup.entries.flatMap { group ->
+      NotificationChannel.entries.map { channel ->
+        dbData.find { it.group == group && it.channel == channel } ?: defaultSettings(user, group, channel)
+      }
+    }
+  }
+
+  fun getSettingValue(
+    notification: Notification,
+    channel: NotificationChannel,
+  ) = notificationSettingRepository
+    .findByUserIdAndGroupAndChannel(
+      notification.user.id,
+      notification.type.group,
+      channel,
+    )?.enabled ?: true
+
+  private fun defaultSettings(
+    user: UserAccount,
+    group: NotificationTypeGroup,
+    channel: NotificationChannel,
+  ) = NotificationSetting().apply {
```

Comment:
We want to make the default for each channel configurable via server config. Please add such option and test it.